wordpress-mobile/WordPress-Android
# Issue/16334 link blogging reminders with prompts onboarding

Fixes #16334

This PR links adds the little help icon (?) to the blogging prompt switch at the blogging reminder cadence setting screen, that leads to the version of onboarding dialog with only "Got it" button that dismisses the dialog.

I did a bit of refactoring to the dialog to accommodate two different variants of the dialog, and left some comments in the diff.
@renanferrari since you had the vision for original code, please let me know if you want me to do anything in a different way :)

[![Image from Gyazo](https://i.gyazo.com/b95055e85d8953caf681c77bad3363c8.gif)](https://gyazo.com/b95055e85d8953caf681c77bad3363c8)

To test:
- Enable blogging prompt FF.
- Navigate to Settings, Blogging Reminder screen.
- Select a date if not selected already.
- Tap on the ? icon near blogging prompt switch.
- Confirm that the onboarding dialog appears, with only one "Got it" button that dismisses it.
- Confirm that the buttons in regular variant of onboarding dialog work as expected.

## Regression Notes
1. Potential unintended areas of impact
Onboarding version of the dialog shows one button.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tests.

3. What automated tests I added (or what prevented me from doing so)
Testing click listeners and the Ready state.

PR submission checklist:
